### PR TITLE
Use PoParser.Core

### PR DESCRIPTION
**Pull request by hishamco**
The aim is to use a new parser for portable objects that uses Parlot behind the scene to make it faster

### CODE REVIEW

File: test/OrchardCore.Tests/Localization/PortableObjectParserTests.cs [hunk 1]
```diff
@@ -100,17 +100,17 @@ public void ParseOnlyTrimsLeadingAndTrailingQuotes()
...
         }
 
-        [Fact]
-        public void ParseHandleUnclosedQuote()
-        {
-            // msgctxt "
-            // msgid "Foo \"{0}\""
-            // msgstr "Foo \"{0}\""
+        //[Fact]
+        //public void ParseHandleUnclosedQuote()
```
Comment (hishamco) [inline test/OrchardCore.Tests/Localization/PortableObjectParserTests.cs:104]: Handle unclosed quote is special implementation I see in OC, we can customize tha `PoParser` to support it out of the box

**Comment by hishamco** [comment]
Benchmarks may added later to measure the PERF

File: test/OrchardCore.Tests/Localization/PortableObjectParserTests.cs (hunk 1 above)
Comment (hishamco) [reply test/OrchardCore.Tests/Localization/PortableObjectParserTests.cs:104]: IMHO this case should be ignore coz it break the syntax, but we could handle it by custom parser implementation

**Comment by hishamco** [comment]
FYI the new parser need some improvement in terms of PERF, for now it expose a rich set of metadata about the parsed PO file as `SyntaxNode`s

**Comment by hishamco** [comment]
@Skrypt you could have a look if you are interested ;)

**Comment by sebastienros** [comment]
How faster is it? Not that it makes sense for parsing PO files to be faster, but that's the only reason invoked.

**Comment by hishamco** [comment]
I don't think it's fast as native parsing, by the new library still need some improvement in terms of PERF, coz I'm still have many question about Parlot - may be I did something not recommended-

The thing that is good about this library is exposing the entire syntax tree, so the developer can easily work with the nodes themselves something similar to Irony, but agian the PERF is crucial

One more thing is related to pasring and Parlot, could you please have a quick look to [SqlParser](https://github.com/hishamco/SqlParser) which I already started in #8215 few months ago, still there are some work to finalize few things remaining in SELECT, but I need your recommendation in case something goes wrong

Thanks

**Comment by jptissot** [comment]
I think this should be closed @hishamco

**Comment by hishamco** [comment]
I will close this for now

**Pull request closed without merging**